File: sdk/python/iploop/exceptions.py
"""IPLoop SDK exceptions."""


class IPLoopError(Exception):
    """Base exception for IPLoop SDK."""


class AuthError(IPLoopError):
    """Invalid or missing API key."""


class QuotaExceeded(IPLoopError):
    """Bandwidth quota exceeded."""

    def __init__(self, message=None):
        super().__init__(
            message or "Quota exceeded. Upgrade at https://iploop.io/pricing"
        )


class ProxyError(IPLoopError):
    """Proxy connection failed after all retries."""


class TimeoutError(IPLoopError):
    """All retries timed out."""

File: sdk/node/src/exceptions.js
class IPLoopError extends Error {
  constructor(message) { super(message); this.name = 'IPLoopError'; }
}
class AuthError extends IPLoopError {
  constructor(message) { super(message || 'Invalid or missing API key'); this.name = 'AuthError'; }
}
class QuotaExceeded extends IPLoopError {
  constructor(message) { super(message || 'Quota exceeded. Upgrade at https://iploop.io/pricing'); this.name = 'QuotaExceeded'; }
}
class ProxyError extends IPLoopError {
  constructor(message) { super(message || 'Proxy connection failed'); this.name = 'ProxyError'; }
}
class TimeoutError extends IPLoopError {
  constructor(message) { super(message || 'Request timed out'); this.name = 'TimeoutError'; }
}
module.exports = { IPLoopError, AuthError, QuotaExceeded, ProxyError, TimeoutError };
